parsl/executors/status_handling.py

```diff
@@ -52,25 +52,25 @@ def status_polling_interval(self):
             return self._provider.status_polling_interval
 
     @abstractmethod
-    def _get_job_ids(self) -> List[object]:
+    def _get_block_and_job_ids(self) -> Tuple[List[str], List[Any]]:
         raise NotImplementedError("Classes inheriting from StatusHandlingExecutor must implement "
-                                  "_get_job_ids()")
+                                  "_get_block_and_job_ids()")
 
-    def _fail_job_async(self, job_id: Any, message: str):
+    def _fail_job_async(self, block_id: Any, message: str):
         """Marks a job that has failed to start but would not otherwise be included in status()
         as failed and report it in status()
         """
-        if job_id is None:
-            job_id = "failed-block-{}".format(self._generated_job_id_counter)
-            self._generated_job_id_counter += 1
-        self._simulated_status[job_id] = JobStatus(JobState.FAILED, message)
+        if block_id is None:
+            block_id = "failed-block-{}".format(self._generated_block_id_counter)
+            self._generated_block_id_counter += 1
+        self._simulated_status[block_id] = JobStatus(JobState.FAILED, message)
 
     def status(self) -> Dict[object, JobStatus]:
```

Comment:
status returns a dict of block IDs -> job statuses here, now, I think?

Reply:
Okay, tried to fix all the `status` instances in commit https://github.com/Parsl/parsl/pull/1952/commits/8e3e23c366b8872845abbcf57329537e3aa4f43e